### DISCUSSION TITLE
Delays Between Job Executions

### DISCUSSION THREAD

**Discussion by arefrafei92** [OptimalBits/bull#2708] [Q&A]
I intend to create a sequence where each job is executed with a delay in between, following a pattern like this:

`Delay(3000) -> Job 1 -> Delay(3000) -> Job 2 -> Delay(3000) -> Job 3 -> ...`

However, the current behavior I'm experiencing seems to introduce the initial delay and then execute all subsequent jobs without further delays:

`Delay(3000) -> Job 1 -> Job 2 -> Job 3 -> ...`

I would like to seek your advice on how to achieve the desired behavior of introducing delays between each job execution consistently.

```
@Injectable()
@Processor('Product')
export class ProductService implements OnModuleInit {
 
  constructor(
    @InjectQueue('Product') private readonly _Q: Queue,
  ){}

  @Process()
  async consumeQ(job: Job<any>) {
      this.logger.debug(job.id);
  }

  async onModuleInit() {
    for (let index = 0; index < 10; index++) {
      const job = await this._Q.add(
       {
          delay: 3000,
          removeOnComplete: true,
          removeOnFail: 3,
          backoff: {
            type: 'exponential',
            delay: 1500,
          },
        },);
    }
  }
}
```
## Bull version
```
 "@nestjs/bull": "^10.1.0"
 "bull": "^4.12.2"
```

**Comment by manast** [comment]
You are adding 10 delayed jobs as soon as the Module initializes. Why aren't you adding the next job inside the consumeQ function?

**Reply by arefrafei92** [reply]
This is just for testing, but I have a function that loads a list of requests and then I want to add all of them into the queue. in that case, every time a job was executed it would add another job, so I had to keep a list of jobs somewhere and then go through it in consumeQ. But my question is why delay isn't working for each job? is it just for initial running?
